lib/BattleHex.h

```diff
@@ -31,17 +31,13 @@ struct DLL_LINKAGE BattleHex //TODO: decide if this should be changed to class f
 	si16 getX() const;
 	si16 getY() const;
 	std::pair<si16, si16> getXY() const;
-	//moving to direction
-	BattleHex& moveInDir(EDir dir, bool hasToBeValid = true);
-	BattleHex& operator+=(EDir dir); //sugar for above
-	//generates new BattleHex moved by given dir
-	BattleHex movedInDir(EDir dir, bool hasToBeValid = true) const;
+	BattleHex& moveInDirection(EDir dir, bool hasToBeValid = true);
+	BattleHex& operator+=(EDir dir);
+	BattleHex generateHexMovedInDirection(EDir dir, bool hasToBeValid = true) const;
 	BattleHex operator+(EDir dir) const;
 	std::vector<BattleHex> neighbouringTiles() const;
-	//returns info about mutual position of given hexes (-1 - they're distant, 0 - left top, 1 - right top, 2 - right, 3 - right bottom, 4 - left bottom, 5 - left)

```

Comment:
I don't agree, only library need documentation for me, comments make trash in the other code. In the definition you have everything beautifully explained, I have changed it to be so.